### PR TITLE
Adds MKL docs

### PR DESCRIPTION
**Pull request by aturner-epcc**
Initial pass at MKL docs

### CODE REVIEW

**Comment by jsindt** [comment]
Quick question -- is it not possible to unload cray-libsci as part of the MKL module?

File: docs/software-libraries/mkl.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+module remove cray-libsci
+module load mkl
+```
+
+| Language | Compile options | Link options |
+|----------|-|-|
+| Fortran | `-m64  -I"${MKLROOT}/include"` | `-L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_gf_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl` |
+| C/C++ | ` -m64  -I"${MKLROOT}/include"` | `-L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl` |
+
+## Threaded MLK with GCC
```
Comment (juanfrh) [inline docs/software-libraries/mkl.md:44]: "MKL" is mispelled.

Comment (juanfrh) [reply docs/software-libraries/mkl.md:44]: > "MKL" is mispelled.

**Review by juanfrh — CHANGES_REQUESTED** [review]
I've just found a typo. The rest looks good to me.

File: docs/software-libraries/mkl.md (hunk 1 above)
Comment (juanfrh) [inline docs/software-libraries/mkl.md:44]: "MKL" is mispelled.

**Comment by aturner-epcc** [comment]
> Quick question -- is it not possible to unload cray-libsci as part of the MKL module?

Yes, this is a good idea and I will do it.

**Review by juanfrh — APPROVED** [review]
LGTM.

**Comment by markgbeckett** [comment]
Andy - you say that MKL will not work with Cray Compiler Suite. Does it work with AMD? If not, is it easier to say MKL only works with the Gnu Compiler suite.

**Comment by aturner-epcc** [comment]
@markgbeckett Thanks. I have clarified this point to say that only GCC is supported.

**Pull request merged**